libnetwork/iptables/firewalld.go

```diff
@@ -282,6 +282,22 @@ func DelInterfaceFirewalld(intf string) error {
 	return nil
 }
 
+func SetupInterfaceFirewalldZone(bridgeName string, enable bool) error {
+	// Either add or remove the interface from the firewalld zone
+	if firewalldRunning {
+		if enable {
+			if err := AddInterfaceFirewalld(bridgeName); err != nil {
+				return err
+			}
+		} else {
+			if err := DelInterfaceFirewalld(bridgeName); err != nil {
```

Comment:
This is not new, but I'm wondering if this is a bug; should this action be idempotent? What I mean is: should we actually produce an error if the interface _was not part of the zone_ ?
https://github.com/moby/moby/blob/e0da5cb9291982a4c7b3c6b0642eeced4ba9f8ac/libnetwork/iptables/firewalld.go#L266-L275

If it wasn't part of the zone, it feels like we can ignore the error ("we're done; nothing to do here, move on move on.. ").

However, for that to work, we probably need to update `AddInterfaceFirewalld` and `DelInterfaceFirewalld` to return properly types errors, and return a [`errdefs.ErrNotFound`](https://github.com/moby/moby/blob/e0da5cb9291982a4c7b3c6b0642eeced4ba9f8ac/errdefs/defs.go#L3-L6), so that we can handle "not found" errors differently when calling those functions.

Which would look something like;

```go
// Remove bridge interface from the docker zone if it exists, and if firewalld is running.
if err := DelInterfaceFirewalld(bridgeName); err != nil && !errdefs.IsNotFound(err) {
	return err
}
```

Reply:
Not really sure about how the error handling should be done but I'll give it a try.

Reply:
I tried to remove the check for firewalld running, but it resulted in errors: `2023/08/14 19:26:22 http: panic serving @: runtime error: invalid memory address or nil pointer dereference` and many other lines.
I agree that checking if firewalld is running is unnecessary, I just have no idea how to code to remove this dependency.
It took me long enough to figure out how to do what I already did.